### PR TITLE
feat: add OutputParser

### PR DESCRIPTION
**Pull request by HokageM**

### CODE REVIEW

File: src/output/OutputParser.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#include "OutputParser.hpp"
+
+#include <GraphMol/SmilesParse/SmartsWrite.h>
+#include <spdlog/spdlog.h>
+
+namespace coaler {
+    void OutputParser::add_aligned_mols(AlignedMolPair& aligned_mol_pair) {
```
Comment (nikplx) [inline src/output/OutputParser.cpp:7]: This thing does not parse stuff, but it writes stuff right? So `OutputWriter` would be better? Though it is implicit that output is written somehwere. So maybe `MolPairWriter` or smth similar?

File: src/output/OutputParser.hpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+         */
+        void save_molecules_w_scores_in_file(const std::string& file_path);
+
+        /**
+         * Prints the aligned molecules with score inside the log.
+         */
+        void print_in_log_molecules_w_scores();
+
+      private:
+        std::vector<AlignedMolPair> aligned_mols_;
```
Comment (nikplx) [inline src/output/OutputParser.hpp:47]: We should get on the same page here. Some people use `m_member` and some people use `member_`.

File: src/output/OutputParser.hpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        /**
+         * Writes the aligned molecules in an output file.
+         * @param file_path
+         */
+        void save_molecules_w_scores_in_file(const std::string& file_path);
+
+        /**
+         * Prints the aligned molecules with score inside the log.
+         */
+        void print_in_log_molecules_w_scores();
```
Comment (nikplx) [inline src/output/OutputParser.hpp:44]: Why should this be a public function? Logging should be a thing that the class takes care of itself

File: src/output/OutputParser.cpp (hunk 1 above)
Comment (HokageM) [reply src/output/OutputParser.cpp:7]: OutputWriter would be okay for me

File: src/output/OutputParser.hpp (hunk 2 above)
Comment (HokageM) [reply src/output/OutputParser.hpp:47]: I dont See m_member in our Code yet.
I am Open for member_ or Just member.
"m_" is for old people

File: src/output/OutputParser.hpp (hunk 3 above)
Comment (HokageM) [reply src/output/OutputParser.hpp:44]: The Idea is that we have an Option for writing the Output to an File or the log / console.
So it is not the logging If writing the Output in a file

File: src/output/OutputParser.hpp (hunk 2 above)
Comment (HokageM) [reply src/output/OutputParser.hpp:47]: https://micro-os-plus.github.io/develop/naming-conventions/#:~:text=Similar%20to%20member%20functions%2C%20all,usually%20start%20with%20a%20noun.

**Review by mariusrueve — APPROVED** [review]

**Pull request merged**